### PR TITLE
reimplement destroy and format in terms of Lwt_unix, get rid of Sys.c…

### PR DESCRIPTION
**Pull request by hannesm**
…ommand, add tests

### CODE REVIEW

**Comment by samoht** [comment]
just out-of-curiosity, why are everyone seems to start using that library? :-)

**Comment by yomimono** [comment]
I'm using it for mirage-net-pcap , which really likes appending to files 
and reading at offsets.

**Comment by hannesm** [comment]
I use it in tls-demo-server (back last year)... and was motivated by @yomimono changes to put it into a state where I actually feel comfortable with it (removing Sys.command, adding missing tests) :)

File: lib/FS_unix.ml [hunk 1]
```diff
@@ -153,31 +141,57 @@ let listdir {base} path =
...
 
-let write {base} path off buf =
+
+let listdir {base} path =
   let path = Fs_common.resolve_filename base path in
-  create_directory (Filename.dirname path) >>= function
+  list_directory path
+
+let rec remove path =
+  let rec rm dir base p =
```
Comment (dsheets) [inline lib/FS_unix.ml]: Nit: would be nice to call `dir` `~is_dir` or something.

File: lib/FS_unix.ml [hunk 2]
```diff
@@ -153,31 +141,57 @@ let listdir {base} path =
...
+    Lwt_unix.LargeFile.stat full >>= fun stat ->
+    match stat.Lwt_unix.LargeFile.st_kind with
+    | Lwt_unix.S_DIR ->
+      list_directory full >>= (function
+          | `Ok files ->
+            Lwt_list.iter_p (rm true full) files >>= fun () ->
+            if dir then Lwt_unix.rmdir full else Lwt.return_unit
+          | `Error e -> Lwt.fail e)
+    | Lwt_unix.S_REG | Lwt_unix.S_LNK -> Lwt_unix.unlink full
+    | _ -> Lwt.fail (Error (`Unknown_error "cannot remove unknown file type"))
```
Comment (dsheets) [inline lib/FS_unix.ml]: Is this really an Lwt-level error condition and not a ``` `Error ```-level condition?

Comment (hannesm) [inline lib/FS_unix.ml]: rm is wrapper in `try_lwt` down, which now handles `Error` gracefully

**Comment by hannesm** [comment]
@dsheets addressed your comments..

**Pull request merged**